### DISCUSSION TITLE
Language server thinks Actor does not have stop() property

### DISCUSSION THREAD

**Discussion by lincore81** [excaliburjs/Excalibur#3323] [Q&A]
Hey y'all, I can't make up my mind if the type definitions are lacking or if it's me, although I suspect it to be the latter.

I'm following the [flappy bird tutorial (chapter 5)](https://excaliburjs.com/docs/step-5-plumbing-pipes), which includes `stop`ping the bird on collision. And while the method definitely works, the language server thinks it doesn't exist. 

I encountered the issue in neovim and vscode (default tsconfig except target set to esnext). Does anybody know what's going on here?

package.json:
```json
--- snip ---
  "devDependencies": {
    "typescript": "^5.7.2",
    "vite": "^6.0.6"
  },
  "dependencies": {
    "excalibur": "^0.30.2"
  }
```

src/actors/bird/bird.ts:
```ts
import { Actor, clamp, Collider, Color, Engine, Keys, vec } from "excalibur";
import { Ground } from "../ground/ground";
import { Pipe } from "../pipe/pipe";

export class Bird extends Actor {
  jumping = false;

  constructor() {
    super({
      pos: vec(100, 120),
      width: 16,
      height: 16,
      color: Color.Yellow,
    });
  }
  
  // --- SNIP ---

  override onCollisionStart(_: Collider, other: Collider): void {
    if (other.owner instanceof Ground || other.owner instanceof Pipe) {
      this.stop(); // <-- lsp error: Property 'stop' does not exist on type 'Bird'.
    }
  }
}
```

**Comment by eonarheim** [comment]
@lincore81 Thanks for the post! Apologies for this, this is a gap in the tutorial text! I'll update the prose to fix this asap.

`.stop()` is implemented on the `Bird` type

Give the final code a look to cross check any errors/gaps you find

https://github.com/excaliburjs/sample-excalibird/blob/main/src/bird.ts#L97

**Reply by eonarheim** [reply]
@lincore81 I added some more clarification to the docs! https://github.com/excaliburjs/Excalibur/commit/f0d9a6ba1740a32df101ba22cd7f199adf251631

Let me know if you have any other recommendations
